src/lib.rs

```diff
@@ -0,0 +1,269 @@
+                        use MemtestError::*;
+                        use MemtestOutcome::*;
+                        match (acc, result) {
+                            (Err(Unknown), _) | (_, Err(Unknown)) => Err(Unknown),
+                            (Err(Timeout), _) | (_, Err(Timeout)) => Err(Timeout),
+                            (Ok(Fail(addr)), _) | (_, Ok(Fail(addr))) => Ok(Fail(addr)),
+                            _ => Ok(Pass),
+                        }
+                    })
+            } else {
+                test(self.base_ptr, memcount, time_left)
+            };
+
+            reports.push(MemtestReport::new(*test_type, test_result));
+        }
+
+        // Unlock memory
+        drop(lockguard);
+
+        #[cfg(windows)]
+        if let Some((min_set_size, max_set_size)) = working_set_sizes {
+            win_working_set::restore_set_size(min_set_size, max_set_size)?;
+        }
+
+        Ok(MemtestReportList {
+            tested_memsize,
+            mlocked,
+            reports,
+        })
+    }
+
+    fn memory_resize_and_lock(&mut self) -> Result<region::LockGuard, ()> {
+        const WIN_OUTOFMEM_CODE: usize = 1453;
+        loop {
+            match region::lock(self.base_ptr, self.memsize) {
+                Ok(lockguard) => {
+                    return Ok(lockguard);
+                }
+                // TODO: macOS error?
+                Err(region::Error::SystemCall(err))
+                    if ((matches!(err.kind(), ErrorKind::OutOfMemory)
+                        || err
+                            .raw_os_error()
+                            .is_some_and(|e| e as usize == WIN_OUTOFMEM_CODE))
+                        && self.allow_mem_resize) =>
+                {
+                    match self.memsize.checked_sub(PAGE_SIZE) {
+                        Some(new_memsize) => self.memsize = new_memsize,
+                        None => return Err(()),
+                    }
+                }
+                _ => {
+                    return Err(());
+                }
+            }
+        }
+    }
+}
+
+#[cfg(windows)]
+mod win_working_set {
+    use super::MemtesterError;
+    use windows::Win32::System::Threading::{
+        GetCurrentProcess, GetProcessWorkingSetSize, SetProcessWorkingSetSize,
+    };
+
+    pub(super) unsafe fn replace_set_size(
+        memsize: usize,
+    ) -> Result<(usize, usize), MemtesterError> {
+        let (mut min_set_size, mut max_set_size) = (0, 0);
+        match GetProcessWorkingSetSize(GetCurrentProcess(), &mut min_set_size, &mut max_set_size) {
+            Ok(()) => (),
+            Err(_) => return Err(MemtesterError::WindowsWorkingSetFailure),
+        }
+        // TODO: Not sure what the best choice of min and max should be
+        match SetProcessWorkingSetSize(
+            GetCurrentProcess(),
+            memsize.saturating_mul(2),
+            memsize.saturating_mul(4),
+        ) {
```

Comment:
As commented, not sure what the best choice of min and max should be. Windows doesn't have a system limit of these values. 
The only consideration I had was that the most amount of memory I can lock is the min set size minus a small overhead (see [Windows documentation](https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtuallock#remarks)) and so it seemed safer to set the min to `memsize * 2`?